Add support for vLLM-Ascend

### Reminder

- [x] I have read the above rules and searched the existing issues.

### Description

I am about to add support for VLLM-ascend.
Interested developers can work together to add support for vllm-ascend.

### Pull Request

_No response_

> Yea how about the result after adding this env?
> 
ine 9 in [59e12bf](/hiyouga/LLaMA-Factory/commit/59e12bffe8d823eaa2ece2d0104379865e60a58d)
>  ENV VLLM_WORKER_MULTIPROC_METHOD=spawn

After exporting VLLM_WORKER_MULTIPROC_METHOD=spawn, the above problem disappeared, but a new problem appeared (as shown below). However, I am not sure whether this is a common problem or a problem with my environment. Next, I will continue to deal with the ERROR CODE: 500001 problem.
```
(VllmWorkerProcess pid=4228) ERROR 03-26 03:24:51 multiproc_worker_utils.py:242] Exception in worker VllmWorkerProcess while processing method init_device.
(VllmWorkerProcess pid=4228) ERROR 03-26 03:24:51 multiproc_worker_utils.py:242] Traceback (most recent call last):
(VllmWorkerProcess pid=4228) ERROR 03-26 03:24:51 multiproc_worker_utils.py:242]   File "/usr/local/python3.10/lib/python3.10/site-packages/vllm/executor/multiproc_worker_utils.py", line 236, in _run_worker_process
(VllmWorkerProcess pid=4228) ERROR 03-26 03:24:51 multiproc_worker_utils.py:242]     output = run_method(worker, method, args, kwargs)
(VllmWorkerProcess pid=4228) ERROR 03-26 03:24:51 multiproc_worker_utils.py:242]   File "/usr/local/python3.10/lib/python3.10/site-packages/vllm/utils.py", line 2196, in run_method
(VllmWorkerProcess pid=4228) ERROR 03-26 03:24:51 multiproc_worker_utils.py:242]     return func(*args, **kwargs)
(VllmWorkerProcess pid=4228) ERROR 03-26 03:24:51 multiproc_worker_utils.py:242]   File "/usr/local/python3.10/lib/python3.10/site-packages/vllm_ascend/worker/worker.py", line 164, in init_device
(VllmWorkerProcess pid=4228) ERROR 03-26 03:24:51 multiproc_worker_utils.py:242]     NPUPlatform.set_device(self.device)
(VllmWorkerProcess pid=4228) ERROR 03-26 03:24:51 multiproc_worker_utils.py:242]   File "/usr/local/python3.10/lib/python3.10/site-packages/vllm_ascend/platform.py", line 90, in set_device
(VllmWorkerProcess pid=4228) ERROR 03-26 03:24:51 multiproc_worker_utils.py:242]     torch.npu.set_device(device)
(VllmWorkerProcess pid=4228) ERROR 03-26 03:24:51 multiproc_worker_utils.py:242]   File "/usr/local/python3.10/lib/python3.10/site-packages/torch_npu/npu/utils.py", line 83, in set_device
(VllmWorkerProcess pid=4228) ERROR 03-26 03:24:51 multiproc_worker_utils.py:242]     torch_npu._C._npu_setDevice(device_id)
(VllmWorkerProcess pid=4228) ERROR 03-26 03:24:51 multiproc_worker_utils.py:242] RuntimeError: SetPrecisionMode:build/CMakeFiles/torch_npu.dir/compiler_depend.ts:155 NPU function error: at_npu::native::AclSetCompileopt(aclCompileOpt::ACL_PRECISION_MODE, precision_mode), error code is 500001
(VllmWorkerProcess pid=4228) ERROR 03-26 03:24:51 multiproc_worker_utils.py:242] [ERROR] 2025-03-26-03:24:51 (PID:4228, Device:0, RankID:-1) ERR00100 PTA call acl api failed
(VllmWorkerProcess pid=4228) ERROR 03-26 03:24:51 multiproc_worker_utils.py:242] [Error]: The internal ACL of the system is incorrect.
(VllmWorkerProcess pid=4228) ERROR 03-26 03:24:51 multiproc_worker_utils.py:242]         Rectify the fault based on the error information in the ascend log.
```

--- a/examples/README.md
+++ b/examples/README.md
@@ -181,6 +181,7 @@ llamafactory-cli export examples/merge_lora/llama3_full_sft.yaml
 #### Batch Generation using vLLM Tensor Parallel
 
 ```
+export VLLM_WORKER_MULTIPROC_METHOD=spawn
 python scripts/vllm_infer.py --model_name_or_path path_to_merged_model --dataset alpaca_en_demo
 ```
 

--- a/examples/README_zh.md
+++ b/examples/README_zh.md
@@ -181,6 +181,7 @@ llamafactory-cli export examples/merge_lora/llama3_full_sft.yaml
 #### 使用 vLLM+TP 批量推理
 
 ```
+export VLLM_WORKER_MULTIPROC_METHOD=spawn
 python scripts/vllm_infer.py --model_name_or_path path_to_merged_model --dataset alpaca_en_demo
 ```
 

--- a/src/llamafactory/cli.py
+++ b/src/llamafactory/cli.py
@@ -17,16 +17,7 @@ import subprocess
 import sys
 from enum import Enum, unique
 
-from . import launcher
-from .api.app import run_api
-from .chat.chat_model import run_chat
-from .eval.evaluator import run_eval
 from .extras import logging
-from .extras.env import VERSION, print_env
-from .extras.misc import find_available_port, get_device_count, is_env_enabled, use_ray
-from .train.tuner import export_model, run_exp
-from .webui.interface import run_web_demo, run_web_ui
-
 
 USAGE = (
     "-" * 70
@@ -41,18 +32,6 @@ USAGE = (
     + "|   llamafactory-cli webui: launch LlamaBoard                        |\n"
     + "|   llamafactory-cli version: show version info                      |\n"
     + "-" * 70
-)
-
-WELCOME = (
-    "-" * 58
-    + "\n"
-    + f"| Welcome to LLaMA Factory, version {VERSION}"
-    + " " * (21 - len(VERSION))
-    + "|\n|"
-    + " " * 56
-    + "|\n"
-    + "| Project page: https://github.com/hiyouga/LLaMA-Factory |\n"
-    + "-" * 58
 )
 
 logger = logging.get_logger(__name__)
@@ -71,8 +50,28 @@ class Command(str, Enum):
     VER = "version"
     HELP = "help"
 
-
 def main():
+    from . import launcher
+    from .api.app import run_api
+    from .chat.chat_model import run_chat
+    from .eval.evaluator import run_eval
+    from .extras.env import VERSION, print_env
+    from .extras.misc import find_available_port, get_device_count, is_env_enabled, use_ray
+    from .train.tuner import export_model, run_exp
+    from .webui.interface import run_web_demo, run_web_ui
+
+    WELCOME = (
+        "-" * 58
+        + "\n"
+        + f"| Welcome to LLaMA Factory, version {VERSION}"
+        + " " * (21 - len(VERSION))
+        + "|\n|"
+        + " " * 56
+        + "|\n"
+        + "| Project page: https://github.com/hiyouga/LLaMA-Factory |\n"
+        + "-" * 58
+    )
+
     command = sys.argv.pop(1) if len(sys.argv) != 1 else Command.HELP
     if command == Command.API:
         run_api()
@@ -128,4 +127,6 @@ def main():
 
 
 if __name__ == "__main__":
+    from multiprocessing import freeze_support
+    freeze_support()
     main()

